### DISCUSSION TITLE
Error Suppression?

### DISCUSSION THREAD

**Discussion by sjstraw** [stoplightio/spectral#1791] [Q&A] [answered]
Hi, all - 

We'd love to apply Spectral to help our devs use our API style guide for new development.  But we aren't starting from scratch; there will be a bunch of APIs which won't adhere to our rules at first, so we will need to go through the process of recreating those APIs and deprecating the old ones.

As this goes on, we will have to tolerate specific non-conforming APIs for awhile while catching new APIs which don't conform.

What facilities does Spectral have that would allow us to suppress errors on specific APIs, so that CI passes, while we clean things up over time?

The docs for Spectral don't seem to mention suppression.  However, it might be possible to define rules or functions which intentionally ignore some content.  What's the best way to do this?

**Comment by P0lip** [comment, marked as answer]
Hey!
Have you considered using [overrides](https://github.com/stoplightio/spectral/blob/develop/docs/guides/4-custom-rulesets.md#overrides)?
I believe this might be a perfect fit for your use case.

**Reply by sjstraw** [reply]
Yes, this seems like exactly what I was looking for! 

Thanks much for pointing me in the right direction!

Do these work with the standard rule sets as well with custom rule sets?

**Reply by P0lip** [reply]
Yes, they work for both the standard rulesets as well as the custom rulesets.
We've just updated docs and added more examples https://github.com/stoplightio/spectral/blob/29576f0fd75faebc4087c23e3746450c54dda0f9/docs/guides/4-custom-rulesets.md#overrides

**Comment by sjstraw** [comment]
Nice!  Thanks, Jakub!
